src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -338,6 +354,12 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
                 }
                 return true;
             }
+            catch (PathTooLongException)
```

Comment:
As I am unable to contact you, I'm going to merge for now.  The reason I've given up on solving the path too long exceptions here is that it seems quite possible on a messed up machine for RoboCopy.exe to never return, leading to a build failure.  I'd rather take my chances with a single bad type of build filling the disk than guaranteed timeout and build failure from this.